platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCache.kt

```diff
@@ -240,7 +240,11 @@ class DefaultConfigurationCache internal constructor(
             // Can reuse the cache entry for the rest of this build invocation
             cacheAction = ConfigurationCacheAction.LOAD
         }
-        scopeRegistryListener.dispose()
```

Comment:
My feedback from the original PR: https://github.com/gradle/gradle/pull/30546/files#r1759395352